CONTRIBUTING.md

```diff
@@ -109,42 +115,346 @@ Check out [building instructions](https://github.com/prestodb/presto/tree/master
+      Co-authored-by: Foo Bar <foo@bar.com>
+      ```
+    * Try to follow the original PR/commits structure. Do not mingle multiple original commits or reorder them.
+    * Do not add big modifications in the same backported commit, unless it’s the minimum required to resolve conflicts.
+    * If needed, add new separate commits following the backport commit for necessary changes.
+
+
 
 ## Committers
 
 Presto committers are defined as [code owners](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) and documented in the project's [`CODEOWNERS`](CODEOWNERS) file.  Each line in the `CODEOWNERS` file defines a module or submodule that the committer has the rights to approve.  New modules and submodules for CODEOWNERS may be added as needed.
 
 New committers are approved by majority vote of the TSC ([see TSC charter](https://github.com/prestodb/tsc/blob/master/CHARTER.md)).  To become a committer, reach out to an [existing TSC member](https://github.com/prestodb/tsc#members) and ask for their feedback on your eligibility (see: [How to become a Presto Committer?](https://github.com/prestodb/presto/wiki/How-to-become-a-Presto-committer%3F)).  Note: to expedite the process, consider creating a document that outlines your Github stats, such as the number of reviews, lines of code added, number of PRs, and outlines particularly outstanding code and review contributions.  If the TSC member believes you are eligible, they will submit your nomination to a vote by the TSC, typically in the form of a PR that adds your handle to the `CODEOWNERS` file.  The process is complete once the PR is merged.
 
+## Pull Requests
+* #### PR size and structure
+    * A PR can consist of multiple small commits, preferably not exceeding 20(?).
+    * The total number of lines modified in a single PR shall not exceeding 5000(?)
+    * The commits sequence shall be in the dependencies order, not in date created/modified order
+    * Every commit in the PR shall pass all the tests
+      Example: https://github.com/prestodb/presto/pull/12991/commits
+
+* #### PR title
+    * Please refer to the commit title
+* #### PR message
+    * Tests
+    * Release notes
+    * Metadata
+        * Resolves #1234
+* #### Backport PR
+    * Must refer to the original PR and provide link in the PR message
+    * Must refer to original authors in the PR message
+    * Do NOT change original commits titles or messages.
+    * If original PR contains multiple commits, follow the same structure.
+
+
 ## Pull Request Checklist
 
-- Branch from the master branch and, if needed, rebase to the current master
-  branch before submitting your pull request. If it doesn't merge cleanly with
-  master you may be asked to rebase your changes.
+If you are not yet an official committer, you must fork from Master and then submit a Pull Request to contribute as an individual.
+
+If you have commit permissions:
```

Comment:
I think this can be simplified, since even committers must have a fork and then submit a PR from their fork.

It might make sense to link to the fork and pull model, since that is what we follow: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/getting-started/about-collaborative-development-models#fork-and-pull-model

Reply:
added and adjusted some language